tests/NSubstitute.Acceptance.Specs/FieldReports/Issue282_MultipleReturnValuesParallelism.cs

```diff
@@ -20,10 +22,13 @@ public void ReturnsMultipleValuesInParallel()
             var substitute = Substitute.For<IFoo>();
             substitute.Foo().Returns(ret1, ret2);
 
-            var runningTask1 = Task.Run(() => substitute.Foo());
-            var runningTask2 = Task.Run(() => substitute.Foo());
+            var runningTask1 = Task.Factory.StartNew(() => substitute.Foo());
+            var runningTask2 = Task.Factory.StartNew(() => substitute.Foo());
+            var tasks = new[] {runningTask1, runningTask2};
 
-            var results = Task.WhenAll(runningTask1, runningTask2).Result;
+            var results = new List<string>();
+            Task.Factory.StartNew(() => Task.WaitAll(tasks));
+            results.AddRange(tasks.Select(t => t.Result));
```

Comment:
Open to suggestions about nicer ways to do this? Working mainly with .NET 4.0 - afraid I'm not very familiar with the async API!

Reply:
@ChrisMaddock I believe you want simply `TaskEx.WhenAll` which will have to be #if'd.

Reply:
(Once you get Bcl.Async working.)

Reply:
Great - thanks!